test/integration/socketio.test.js

```diff
@@ -25,54 +25,83 @@ describe('Socket.io authentication', function() {
   });
 
   it('returns not authenticated error for protected endpoint', done => {
+    socket.emit('todos::get', 'laundry', e => {
+      delete e.stack;
+      delete e.type;
+
+      expect(e).to.deep.equal({
+        name: 'NotAuthenticated',
+        message: 'You are not authenticated.',
+        code: 401,
+        className: 'not-authenticated',
+        errors: {}
+      });
+      done();
+    });
+  });
+
+  it('creates a valid token using the `authenticate` event', done => {
     socket.emit('authenticate', {
       login: 'testing'
-    }, function() {
-      console.log(arguments);
+    }, function(error, data) {
+      expect(error).to.not.be.ok;
+      expect(data.token).to.exist;
       done();
     });
   });
 
-  // it('creates a valid token via HTTP with custom auth', () => {
-  //   return request({
-  //     url: '/authentication',
-  //     method: 'POST',
-  //     body: {
-  //       login: 'testing'
-  //     }
-  //   }).then(body => {
-  //     expect(body.token).to.exist;
-  //     return app.authentication.verifyJWT(body);
-  //   }).then(verifiedToken => {
-  //     const p = verifiedToken.payload;
-
-  //     expect(p).to.exist;
-  //     expect(p.iss).to.equal('feathers');
-  //     expect(p.userId).to.equal(0);
-  //     expect(p.authentication).to.equal('test-auth');
-  //   });
-  // });
-
-  // it('allows access to protected service with valid token, populates user', () => {
-  //   return request({
-  //     url: '/authentication',
-  //     method: 'POST',
-  //     body: {
-  //       login: 'testing'
-  //     }
-  //   }).then(login =>
-  //     request({
-  //       url: '/todos/dishes',
-  //       headers: {
-  //         'Authorization': login.token
-  //       }
-  //     })
-  //   ).then(data => {
-  //     expect(data).to.deep.equal({
-  //       id: 'dishes',
-  //       description: 'You have to do dishes',
-  //       user: { id: 0, name: 'Tester' }
-  //     });
-  //   });
-  // });
+  it('authenticated socket allows accesss and populates user', done => {
+    socket.emit('authenticate', {
+      login: 'testing'
+    }, function(error) {
+      if(error) {
+        return done(error);
+      }
+
+      socket.emit('todos::get', 'laundry', function(error, data) {
+        expect(data).to.deep.equal({
+          id: 'laundry',
+          description: 'You have to do laundry',
+          user: { id: 0, name: 'Tester' }
+        });
+        done();
+      });
+    });
+  });
+
+  it('app login and logout events', done => {
```

Comment:
I typically still break them up into one `expect` per `it` block mostly because if you change the result coming back or remove an attribute it's easier to determine which test needs to change.

However, that only applies if the setup cost is pretty negligible, which it might not be in this particular case.